src/file/tokio_compat_file.rs

```diff
@@ -463,12 +525,36 @@ impl AsyncWrite for TokioCompatFile {
         // sftp v3 cannot send more than self.max_write_len() data at once.
         let max_write_len = self.max_write_len_impl();
 
-        let n: u32 = buf
+        let mut n: u32 = buf
             .len()
             .try_into()
             .map(|n| min(n, max_write_len))
             .unwrap_or(max_write_len);
 
+        let write_limit = self.get_auxiliary().tokio_compat_file_write_limit();
+        let mut write_len = self.write_len;
+
+        if write_len == write_limit {
+            ready!(self.as_mut().flush_one(cx))?;
+            write_len = self.write_len;
+        }
+
+        let new_write_len = match write_len.checked_add(n as usize) {
+            Some(new_write_len) if new_write_len > write_limit => {
+                n = (write_limit - write_len) as u32;
```

Comment:
```suggestion
                n = (write_limit - write_len).try_into().unwrap();
```

I would like the same to apply to other assignments of n, too